common/sketch/src/main/java/org/apache/spark/util/sketch/BloomFilterImpl.java

```diff
@@ -193,14 +193,29 @@ public boolean isCompatible(BloomFilter other) {
 
   @Override
   public BloomFilter mergeInPlace(BloomFilter other) throws IncompatibleMergeException {
+    BloomFilterImpl otherImplInstance = checkCompatibilityForMerge(other);
+
+    this.bits.putAll(otherImplInstance.bits);
+    return this;
+  }
+
+  @Override
+  public BloomFilter intersectInPlace(BloomFilter other) throws IncompatibleMergeException {
+    BloomFilterImpl otherImplInstance = checkCompatibilityForMerge(other);
+
+    this.bits.and(otherImplInstance.bits);
+    return this;
+  }
+
+  private BloomFilterImpl checkCompatibilityForMerge(BloomFilter other) throws IncompatibleMergeException {
     // Duplicates the logic of `isCompatible` here to provide better error message.
     if (other == null) {
       throw new IncompatibleMergeException("Cannot merge null bloom filter");
     }
 
     if (!(other instanceof BloomFilterImpl)) {
       throw new IncompatibleMergeException(
-        "Cannot merge bloom filter of class " + other.getClass().getName()
+              "Cannot merge bloom filter of class " + other.getClass().getName()
```

Comment:
nit: seem to be unnecessary changed indentation.

Reply:
reverted. thanks!